imagebuildah/stage_executor.go

```diff
@@ -1092,15 +1093,8 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 			noRunsRemaining = !ib.RequiresStart(&parser.Node{Children: children[i+1:]})
 		}
 
-		// If we're doing a single-layer build, just process the
-		// instruction.
-		if !s.executor.layers {
-			err := ib.Run(step, s, noRunsRemaining)
-			if err != nil {
-				logrus.Debugf("Error building at step %+v: %v", *step, err)
-				return "", nil, fmt.Errorf("error building at STEP \"%s\": %w", step.Message, err)
-			}
-			// In case we added content, retrieve its digest.
+		// Get content summary for cases where we added content, retrieve its digest.
+		getContentSummaryAfterAddingContent := func() string {
```

Comment:
Please turn this into an actual top-level function
  - That makes it, at least in principle, unit-testable
  - It would actually make `Execute` a bit shorter.

(Possibly even method on `CompositeDigester`, but I’m not sure about the debug log output in that case… maybe a method in `stageExecutor` is a viable compromise.)